RecoLocalMuon/GEMSegment/plugins/MuonSegFit.cc

```diff
@@ -0,0 +1,470 @@
+// ------------------------- //
+// MuonSegFit.cc 
+// Created:  11.05.2015
+// Based on CSCSegFit.cc
+// ------------------------- //
+
+#include "RecoLocalMuon/GEMSegment/plugins/MuonSegFit.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+
+bool MuonSegFit::fit(void) {
+  if ( fitdone() ) return fitdone_; // don't redo fit unnecessarily
+  short n = nhits();
+  switch ( n ) {
+  case 1:
+    edm::LogVerbatim("MuonSegFit") << "[MuonSegFit::fit] - cannot fit just 1 hit!!";
+    break;
+  case 2:
+    fit2();
+    break;
+  case 3:
+  case 4:
+  case 5:
+  case 6:
+    fitlsq();
+    break;
+  default:
+    edm::LogVerbatim("MuonSegFit") << "[MuonSegFit::fit] - cannot fit more than 6 hits!!"; // should make this :: "cannot fit more than 4 hits!!"
+  }  
+  return fitdone_;
+}
+
+void MuonSegFit::fit2(void) {
+
+  // Just join the two points
+  // Equation of straight line between (x1, y1) and (x2, y2) in xy-plane is
+  //       y = mx + c
+  // with m = (y2-y1)/(x2-x1)
+  // and  c = (y1*x2-x2*y1)/(x2-x1)
+  //
+  // Now we will make two straight lines
+  // one in xz-plane, another in yz-plane
+  //       x = uz + c1 
+  //       y = vz + c2
+
+  edm::LogVerbatim("MuonSegFit") << "[MuonSegFit:fit2]-------------------------------------";
```

Comment:
This will be called many times per event, which is rather costly.
Please switch to LogDebug or hide behind ``` if (debug ) ```